Cilsil.Test/Assets/TestClass.cs

```diff
@@ -37,14 +37,9 @@ public struct TestStruct
 
         public TestClass() { }
 
-        public TestClass(TestClass testClass)
+        public TestClass(string filename)
```

Comment:
I think it's better to actually create an optional string parameter (default null); if a non-null string is provided, the InstanceStreamReaderField is initialized with the string.

Reply:
> I think it's better to actually create an optional string parameter (default null); if a non-null string is provided, the InstanceStreamReaderField is initialized with the string.

I found that the function invoked `TestClass(TestClass testClass)` (`InvokeConstructorWithNullParam`) was not used anywhere. So I removed `InvokeConstructorWithNullParam` function and reused the constructor as `TestClass(string filename)`.

Reply:
I'm pretty sure this method was added (with the method InvokeConstructorWithNullParam()) to provide coverage over a case in which we were providing arguments in a flipped order, and where we were thus seeing cases where "this" was null. The methods of TestClass are often not for actual explicit invocation (as you see with InitializeInstanceObjectField, for example) but to provide coverage over certain constructs. I don't think we should remove this case -- i believe it was included to validate against a regression.

Reply:
Done. Not exactly like you suggested though. If we do it like you suggested, Infer will think it is possible that `string filename` is not null and there will be a resource leak inside `InvokeConstructorWithNullParam()`.